store/storetest/user_store.go

```diff
@@ -1671,33 +1690,33 @@ func testUserStoreGetByUsername(t *testing.T, ss store.Store) {
 	defer func() { require.Nil(t, ss.Bot().PermanentDelete(u3.Id)) }()
 
 	t.Run("get u1 by username", func(t *testing.T) {
-		result, err := ss.User().GetByUsername(u1.Username)
-		require.Nil(t, err)
-		assert.Equal(t, u1, result)
+		result := <-ss.User().GetByUsername(u1.Username)
+		require.Nil(t, result.Err)
+		assert.Equal(t, u1, result.Data.(*model.User))
 	})
 
 	t.Run("get u2 by username", func(t *testing.T) {
-		result, err := ss.User().GetByUsername(u2.Username)
-		require.Nil(t, err)
-		assert.Equal(t, u2, result)
+		result := <-ss.User().GetByUsername(u2.Username)
+		require.Nil(t, result.Err)
+		assert.Equal(t, u2, result.Data.(*model.User))
 	})
 
 	t.Run("get u3 by username", func(t *testing.T) {
-		result, err := ss.User().GetByUsername(u3.Username)
-		require.Nil(t, err)
-		assert.Equal(t, u3, result)
+		result := <-ss.User().GetByUsername(u3.Username)
+		require.Nil(t, result.Err)
+		assert.Equal(t, u3, result.Data.(*model.User))
 	})
 
 	t.Run("get by empty username", func(t *testing.T) {
-		_, err := ss.User().GetByUsername("")
-		require.NotNil(t, err)
-		require.Equal(t, err.Id, "store.sql_user.get_by_username.app_error")
+		result := <-ss.User().GetByUsername("")
```

Comment:
this feels like a merge resolution error. This should not have been changed, which is why the tests are failing. @Hyaxia

Reply:
What git commands would you suggest to perform? I cannot merge it again with master, and I guess performing a reset --hard is not ideal either.

Reply:
@Hyaxia i think the changes you want to untangle might be part of a commit so it might not be that easy. I would just suggest restoring the test file to what's in master and just redoing the changes.

Reply:
@ali-farooq0 I have tried to it as clean as possible.
Now I get the same error as before, about shadowing the `err` variable.
Could you check the error and the comment I wrote below and try to understand what is the issue?

Reply:
👀 now